processing/src/main/java/org/apache/druid/query/Query.java

```diff
@@ -110,7 +117,24 @@
    * after it is deserialized. This is because {@link BaseQuery#getContext()} uses
    * {@link QueryContext#getMergedParams()} for serialization, and queries accept a map for deserialization.
    */
-  QueryContext getQueryContext();
+  default QueryContext getQueryContext()
+  {
+    return IS_LEGACY_CONTEXT;
```

Comment:
From looking at the code in this PR, why not return `null` and do null tests?

Reply:
heh, I just left a comment about the same thing here https://github.com/apache/druid/pull/12396#issuecomment-1136547761

I considered this, but decided against it because it didn't seem like actual implementations should legitimately return null and so seemed strange to mark `@Nullable`. Im not very attached to this though, so I could do the `null` thing instead, and mark it as `@Nullable` and update the javadocs to note that its not _actually_ nullable and is only for backwards compatibility and explain the implications.

Reply:
which since you're the 2nd person to suggest it, couple with the fact that i initially considered doing it that way, maybe it would be better?

Reply:
changed to `null`